README.md

```diff
@@ -178,4 +208,49 @@ so for instance...
 ### docker-compose.yml
 In the docker-compose files there are the definitions of where an app should be reached.
 
-Docker-compose will use the variables defined above, and put them under the `labels` section of a container (if a container needs it). The `labels` section is parsed by traefik to route all the HTTP calls to the correct containers.
+Docker-compose will use the variables defined above, and put them under the `labels` section of a container (if a container needs it). The `labels` section is parsed by traefik to route all the HTTP calls to the correct containers. In other words, this is where the values contained in the `.env` file are used to specify that the app replies on '`my.`appserver.test' instead of e.g. '`magic.`appserver.test'.
+
+### /etc/hosts
+
+This file which is located on the _HOST_ machine is used to add new [subdomains for the services](#subdomains-registered-on-traefik): e.g. add `pgadmin` to be able to go to `pgadmin.appserver.test` and use a visual tool for the databases. The important part is that this name of subdomain matches the one defined in the file `.env`
+
+# Troubleshooting / other
+
+- If you are stuck and something doesn't seem to work, make sure you don't have a mismatch between `/etc/hosts`, the URL you type in the browser, and the address that the system expects. See file `current-config.yml` for that: it is a file which is generated every time `make start` is launched. It contains the description of the desired state of the app.
+
+- You can add dev-tools to the array of services and troubleshoot docker using portainer (mentioned above). Make sure that you have added also the URL to the hostfile (`/etc/hosts`) and that it matches the variable defined in `.env`
+
+- If you remove services from array `ENABLED_SERVICES`, they are not stopped. This is not a problem in general, just don't be surprised if when you start (having the configuration `ENABLED_SERVICES=core:frontend:events`) and stop later (with the configuration `ENABLED_SERVICES=events`) you find a message `WARNING: found orphan containers`
+
+## How to run unit/integration tests when you start the system
+
+?? @serge1peshcoff
```

Comment:
@serge1peshcoff a question for you

Reply:
Short answer on how to run tests on hosts for all services except mailer:

(need to do one time)
- install node
- npm install in the app folder
- install postgres (no idea how to do it on win and Linux, on Mac I use Postgres.app. probably docker container will do, check out how I do it in .circleci/config.yml for CI)

Run tests:
- npm test

Run lint:
- npm run lint


Should be enough. @linuxbandit

Reply:
for postgres on Linux, I did it once. I'll look up my history; I did use docker for that, it worked quite well. I'll figure out which config I had to change locally.

Reply:
perhaps we can create a target of `helper.sh` that could e.g. run `--test` ?

Reply:
well it'd suck